diff --git a/node_modules/@hocuspocus/server/dist/hocuspocus-server.cjs b/node_modules/@hocuspocus/server/dist/hocuspocus-server.cjs
index 8a1ba8b..23dc813 100644
--- a/node_modules/@hocuspocus/server/dist/hocuspocus-server.cjs
+++ b/node_modules/@hocuspocus/server/dist/hocuspocus-server.cjs
@@ -1546,8 +1546,9 @@ class Hocuspocus {
             this.hooks('onRequest', { request, response })
                 .then(() => {
                 // default response if all prior hooks don't interfere
-                response.writeHead(200, { 'Content-Type': 'text/plain' });
-                response.end('OK');
+                // REMOVED BECAUSE IT RUNS BEFORE PASSPORT.JS CAN SEND ITS RESPONSE:
+                //response.writeHead(200, { 'Content-Type': 'text/plain' });
+                //response.end('OK');
             })
                 .catch(e => {
                 // if a hook rejects and the error is empty, do nothing
@@ -1560,12 +1561,14 @@ class Hocuspocus {
         });
         server.on('upgrade', (request, socket, head) => {
             this.hooks('onUpgrade', { request, socket, head })
-                .then(() => {
+                .then((shouldHandleUpgrade) => {
                 // let the default websocket server handle the connection if
                 // prior hooks don't interfere
-                websocketServer.handleUpgrade(request, socket, head, ws => {
+                if (shouldHandleUpgrade) {
+                  websocketServer.handleUpgrade(request, socket, head, ws => {
                     websocketServer.emit('connection', ws, request);
-                });
+                  });
+                }
             })
                 .catch(e => {
                 // if a hook rejects and the error is empty, do nothing
diff --git a/node_modules/@hocuspocus/server/dist/hocuspocus-server.esm.js b/node_modules/@hocuspocus/server/dist/hocuspocus-server.esm.js
index 4db9ac9..c55ea12 100644
--- a/node_modules/@hocuspocus/server/dist/hocuspocus-server.esm.js
+++ b/node_modules/@hocuspocus/server/dist/hocuspocus-server.esm.js
@@ -1517,8 +1517,9 @@ class Hocuspocus {
             this.hooks('onRequest', { request, response })
                 .then(() => {
                 // default response if all prior hooks don't interfere
-                response.writeHead(200, { 'Content-Type': 'text/plain' });
-                response.end('OK');
+                // REMOVED BECAUSE IT RUNS BEFORE PASSPORT.JS CAN SEND ITS RESPONSE:
+                //response.writeHead(200, { 'Content-Type': 'text/plain' });
+                //response.end('OK');
             })
                 .catch(e => {
                 // if a hook rejects and the error is empty, do nothing
