diff --git a/node_modules/typescript/lib/tsc.js b/node_modules/typescript/lib/tsc.js
index e144201..97b71d7 100644
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -98851,20 +98851,22 @@ var ts;
     ts.getLocaleTimeString = getLocaleTimeString;
     function createWatchStatusReporter(system, pretty) {
         return pretty ?
-            function (diagnostic, newLine, options) {
-                clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);
-                var output = "[".concat(ts.formatColorAndReset(getLocaleTimeString(system), ts.ForegroundColorEscapeSequences.Grey), "] ");
-                output += "".concat(ts.flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)).concat(newLine + newLine);
-                system.write(output);
+        function (diagnostic, newLine, options) {
+                // suppress logging when watching for changes
+                // clearScreenIfNotWatchingForFileChanges(system, diagnostic, options);
+                // var output = "[".concat(ts.formatColorAndReset(getLocaleTimeString(system), ts.ForegroundColorEscapeSequences.Grey), "] ");
+                // output += "".concat(ts.flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)).concat(newLine + newLine);
+                // system.write(output);
             } :
             function (diagnostic, newLine, options) {
-                var output = "";
-                if (!clearScreenIfNotWatchingForFileChanges(system, diagnostic, options)) {
-                    output += newLine;
-                }
-                output += "".concat(getLocaleTimeString(system), " - ");
-                output += "".concat(ts.flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)).concat(getPlainDiagnosticFollowingNewLines(diagnostic, newLine));
-                system.write(output);
+                // suppress logging when watching for changes
+                // var output = "";
+                // if (!clearScreenIfNotWatchingForFileChanges(system, diagnostic, options)) {
+                //     output += newLine;
+                // }
+                // output += "".concat(getLocaleTimeString(system), " - ");
+                // output += "".concat(ts.flattenDiagnosticMessageText(diagnostic.messageText, system.newLine)).concat(getPlainDiagnosticFollowingNewLines(diagnostic, newLine));
+                // system.write(output);
             };
     }
     ts.createWatchStatusReporter = createWatchStatusReporter;
